modules/perception/obstacle/camera/visualizer/frame_content.h

```diff
@@ -0,0 +1,201 @@
+            return (lhs._timestamp < rhs._timestamp);
+        }
+    }
+};
+
+class FrameContent {
+public:
+    FrameContent();
+    ~FrameContent();
+    enum {
+        PC_CONTINUOUS, IMAGE_CONTINUOUS, NONE_CONTINUOUS
+    };
+    void update_timestamp(double ref);
+    void set_image_content(double timestamp, cv::Mat image);
+    void set_camera_content(double timestamp, Eigen::Matrix4d pose_c2w,
+                            const std::vector<ObjectPtr>& objects,
+                            const CameraFrameSupplement& supplement);
+    void set_camera_content(double timestamp, Eigen::Matrix4d pose_c2w,
+                            const std::vector<ObjectPtr>& objects);
+
+    void set_radar_content(double timestamp, const std::vector<ObjectPtr>& objects);
+    void set_fusion_content(double timestamp, const std::vector<ObjectPtr>& objects);
+    void set_gt_content(double timestamp, const std::vector<ObjectPtr>& objects);
+    void set_camera2car_pose(Eigen::Matrix4d pose_cam2velo);
+
+    void set_motion_content(double timestamp, MotionBufferPtr motion_buffer);
+    Eigen::Matrix4d get_opengl_camera_system_pose();
+    Eigen::Matrix4d get_camera_to_world_pose();
+    Eigen::Matrix4d get_pose_v2w();
+    cv::Mat get_camera_image();
+
+    int get_pose_type() {
+        return _continuous_type;
+    }
+
+    void set_pose_type(int type) {
+        _continuous_type = type;
+    }
+
+    std::vector<ObjectPtr> get_camera_objects();
+    std::vector<ObjectPtr> get_radar_objects();
+    double get_visualization_timestamp();
+
+    inline bool has_radar_data() {
+        return _radar_caches.size();
+    }
+
+    CameraFrameSupplementPtr get_camera_frame_supplement();
+
+    inline bool has_camera_data() {
+        return _camera_caches.size();
+    }
+    /*   inline void set_camera2velo_pose(const Eigen::Matrix4d& pose) {
+           _pose_camera2velo = pose;
+       }*/
+
+    // fused output
+    std::vector<ObjectPtr> get_fused_objects();
+    // gt
+    std::vector<ObjectPtr> get_gt_objects();
+
+    const MotionBuffer get_motion_buffer() ;
+protected:
+    //coordinate transform utilities
+    void offset_object(ObjectPtr object, const Eigen::Vector3d& offset);
+
+private:
+    //input
+    //1.radar
+    std::map<double, RadarContent> _radar_caches;
+    double _current_radar_timestamp;
+
+    //2.camera
+    std::map<double, CameraContent> _camera_caches;
+    double _current_camera_timestamp;
+
+    //3.fusion
+    std::map<double, FusionContent> _fusion_caches;
+    double _current_fusion_timestamp;
+
+    //4.ground truth
+    std::map<double, GroundTruthContent> _gt_caches;
+    double _current_gt_timestamp;
+
+    //5.image
+    std::map<double, ImageContent> _image_caches;
+    double _current_image_timestamp;
+
+    //6.motion
+    std::map<double, MotionContent> _motion_caches;
+    double _current_motion_timestamp;
+
+    Eigen::Vector3d _global_offset;
+    bool _global_offset_initialized;
+    int _continuous_type;
```

Comment:
It seems like that multiple member variables are not set in the constructor.

Reply:
will change later after code is stable